File: leetcode/1.两数之和/1.两数之和.go
package main
/*
1.找到target-num[i]的值，返回index
2.通过mapping把时间复杂度从O(N)降到O(1)
 */
func twoSum(nums []int, target int) []int {
	// 初始化mapping，用于加快查找target-nums的值
	mapping := map[int]int{}
	for i,num := range nums {
		// 如果找到就返回，找不到就添加
		if p, ok := mapping[target-num]; ok{
			return []int{i,p}
		}else{
			mapping[num] = i
		}
	}
	return nil
}

File: hot/leetcode/1.两数之和/1.两数之和.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapping = dict()
        for i in range(len(nums)):
            if mapping.get(target-nums[i]) is not None:
                return [i, mapping[target-nums[i]]]
            mapping[nums[i]] = i
        return []